src/Standards/Generic/Tests/CodeAnalysis/ForLoopWithTestFunctionCallUnitTest.1.inc

```diff
@@ -0,0 +1,45 @@
+<?php
+
+$a = array(1, 2, 3, 4);
+for ($i = 0; $i < count($a); $i++) {
+    $a[$i] *= $i;
+}
+
+for ($i = 0, $c = sizeof($a); $i < $c; ++$i) {
+    $a[$i] *= $i;
+}
+
+$it = new ArrayIterator($a);
+for ($it->rewind(); $it->valid(); $it->next()) {
+    echo $it->current();
+}
+
+for ($i = 0; MyClass::staticMethod($i); $i++) {
```

Comment:
Outside the scope of this PR, but I kind of think that the sniff should **not** trigger for this line (nor for line 21, 39 and 43) as the function call receives `$i` as a parameter and `$i` is the variable being incremented in the third part of the condition.

For now, I think it would be better to just change the parameter passed to the function call in each of these code samples to something other than `$i`.

If you agree that if the variable being in/decremented in the third part of the condition and it is being used as a parameter in the function call in the second part of the condition the sniff, should not flag the function call (or flag it with a different error code, which would allow for excluding the error code), I suggest opening an issue about this.

Reply:
Good catch! I failed to consider this when creating the tests. I just updated them in a new commit.

I tend to agree with you that the behavior of the sniff should be different. and using `$i` in the cases that you highlighted should not trigger the sniff. I will create an issue to continue this discussion as you suggested.